test/trento/application/projectors/cluster_projector_test.exs

```diff
@@ -206,6 +207,31 @@ defmodule Trento.ClusterProjectorTest do
                      1000
   end
 
+  test "should set deregistered_at field to nil when ClusterRestored is received" do
+    insert(:cluster,
+      id: cluster_id = Faker.UUID.v4(),
+      name: name = "deregistered_cluster",
+      selected_checks: [],
+      deregistered_at: DateTime.utc_now()
+    )
+
+    event = ClusterRestored.new!(%{cluster_id: cluster_id})
+
+    ProjectorTestHelper.project(ClusterProjector, event, "cluster_projector")
+    cluster_projection = Repo.get!(ClusterReadModel, event.cluster_id)
+
+    assert nil == cluster_projection.deregistered_at
+
+    assert_broadcast "cluster_registered",
+                     %{
+                       cib_last_written: nil,
```

Comment:
You could add an entry in the `cluster_enrichment`  table for this cluster to see that the enrichment is done properly, and not check just `nil`